contracts/nexus-gateway/src/nexus.rs

```diff
@@ -45,26 +45,47 @@ fn parse_message_id(message_id: &str) -> Result<(nonempty::Vec<u8>, u64), Contra
     Ok((tx_id, index))
 }
 
-impl TryFrom<RouterMessage> for Message {
+impl TryFrom<connection_router::Message> for Message {
     type Error = Report<ContractError>;
 
-    fn try_from(msg: RouterMessage) -> Result<Self, ContractError> {
+    fn try_from(msg: connection_router::Message) -> Result<Self, ContractError> {
         // fallback to using the message ID as the tx ID if it's not in the expected format
         let (source_tx_id, source_tx_index) =
             parse_message_id(&msg.cc_id.id).unwrap_or((msg.cc_id.id.into(), u64::MAX));
 
         Ok(Self {
             source_chain: msg.cc_id.chain,
-            source_address: msg.source_address.clone(),
+            source_address: msg.source_address,
             destination_chain: msg.destination_chain,
-            destination_address: msg.destination_address.clone(),
+            destination_address: msg.destination_address,
             payload_hash: msg.payload_hash,
             source_tx_id,
             source_tx_index,
         })
     }
 }
 
+impl From<Message> for connection_router::Message {
+    fn from(msg: Message) -> Self {
+        Self {
+            cc_id: CrossChainId {
+                chain: msg.source_chain,
+                id: format!(
+                    "{}:{}",
+                    msg.source_tx_id.as_ref().encode_hex::<String>(),
+                    msg.source_tx_index
+                )
```

Comment:
just for my understanding: this doesn't need to get parsed anywhere else right? It only needs to be unique but we could use any kind of format here

Reply:
correct